unison-share-api/src/Unison/Server/Share/Definitions.hs

```diff
@@ -0,0 +1,126 @@
+-- | This module contains implementations of Backend methods which are specialized for Share.
+-- We should likely move them to the Share repository eventually, but for now it's much easier
+-- to ensure they're resilient to refactors and changes in the Backend API if they live here.
+--
+-- Perhaps we'll move them when the backing implementation switches to postgres.
+module Unison.Server.Share.Definitions (definitionForHQName) where
+
+import Control.Lens hiding ((??))
+import Control.Monad.Except
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+import qualified U.Codebase.Causal as V2Causal
+import U.Codebase.HashTags (CausalHash (..))
+import Unison.Codebase (Codebase)
+import qualified Unison.Codebase as Codebase
+import Unison.Codebase.Path (Path)
+import qualified Unison.Codebase.Runtime as Rt
+import qualified Unison.Debug as Debug
+import qualified Unison.HashQualified as HQ
+import qualified Unison.LabeledDependency as LD
+import Unison.Name (Name)
+import Unison.Parser.Ann (Ann)
+import Unison.Prelude
+import qualified Unison.PrettyPrintEnv as PPE
+import qualified Unison.PrettyPrintEnvDecl as PPED
+import qualified Unison.PrettyPrintEnvDecl.Sqlite as PPESqlite
+import Unison.Reference (TermReference)
+import qualified Unison.Reference as Reference
+import qualified Unison.Referent as Referent
+import Unison.Server.Backend hiding (renderDocRefs)
+import qualified Unison.Server.Backend as Backend
+import qualified Unison.Server.Doc as Doc
+import qualified Unison.Server.NameSearch.Sqlite as SqliteNameSearch
+import Unison.Server.Types
+import Unison.Symbol (Symbol)
+import qualified Unison.Syntax.HashQualified as HQ (toText)
+import Unison.Util.Pretty (Width)
+
+-- | Renders a definition for the given name or hash alongside its documentation.
+definitionForHQName ::
```

Comment:
This is a rewrite of `prettyDefinitionsForHQName` but specialized just for Share's backend.

We should likely move them to the Share repository eventually, but for now it's much easier to ensure they're resilient to refactors and changes in the Backend API if they live here.

Perhaps we'll move them when the backing implementation switches to postgres.